theories/algebra/IntDiv.ec

```diff
@@ -351,6 +351,10 @@ proof. by rewrite -mulN1r modzMmr mulN1r. qed.
 lemma nosmt modzBm m n d : (m %% d - n %% d) %% d = (m - n) %% d.
 proof. by rewrite -modzDm -modzNm !modz_mod modzNm modzDm. qed.
 
+lemma modzNDmr m n d :
+  (m - n %% d) %% d = (m - n) %% d.
+proof. smt(modzNm modzDmr). qed.
+
```

Comment:
```suggestion
lemma nosmt modzNDmr m n d : (m - n %% d) %% d = (m - n) %% d.
proof. by rewrite -modzDmr -modzNm !modz_mod modzNm modzDmr. qed.

```

Reply:
Fixed, but I don't understand why. Why do lemmas like this all have `nosmt`, and why is this proof preferable over the smt call?

Reply:
> why is the suggested change preferable over the smt call

Because SMT calls do have a cost in forking out new processes, loading up the SMT solvers' code, saturating axioms and solving, which affects CI times and so on.